cranelift-codegen/src/isa/x86/enc_tables.rs

```diff
@@ -1223,16 +1223,25 @@ fn convert_ineg(
     let mut pos = FuncCursor::new(func).at_inst(inst);
     pos.use_srcloc(inst);
 
+    // FIXME Is it necessary to do this? I thought the generated code only calls this function if
+    // the opcode is ineg, maybe this could just be an assert? I guess this is a very common
+    // pattern in this file, so maybe there's a good reason?
     if let ir::InstructionData::Unary {
         opcode: ir::Opcode::Ineg,
         arg,
     } = pos.func.dfg[inst]
     {
         let value_type = pos.func.dfg.value_type(arg);
-        if value_type.is_vector() && value_type.lane_type().is_int() {
+        let zero_value = if value_type.is_vector() && value_type.lane_type().is_int() {
             let zero_immediate = pos.func.dfg.constants.insert(vec![0; 16].into());
-            let zero_value = pos.ins().vconst(value_type, zero_immediate); // this should be legalized to a PXOR
-            pos.func.dfg.replace(inst).isub(zero_value, arg);
-        }
+            pos.ins().vconst(value_type, zero_immediate) // this should be legalized to a PXOR
+        } else if value_type.is_int() {
+            pos.ins().iconst(value_type, 0)
+        } else {
+            // FIXME Should I panic here? Or maybe add something to the verifier? The ghost error
+            // in the verifier is kind of unintuitive.
+            panic!()
```

Comment:
I think so; I see that type of thing up above. Something like the following would be good to get both the type and the legalization that failed: `panic!("Can't convert ineg of {}", valuetype)`.

Reply:
I added this panic message